collector/indices.go

```diff
@@ -14,44 +14,421 @@
 package collector
 
 import (
+	"context"
 	"encoding/json"
-	"fmt"
-	"io"
 	"log/slog"
 	"net/http"
 	"net/url"
-	"path"
 	"sort"
 	"strconv"
 
 	"github.com/prometheus-community/elasticsearch_exporter/pkg/clusterinfo"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-type labels struct {
-	keys   func(...string) []string
-	values func(*clusterinfo.Response, ...string) []string
-}
+var (
+	indicesDocsPrimary = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "indices", "docs_primary"),
+		"Count of documents with only primary shards",
+		[]string{"index", "cluster"}, nil,
+	)
```

Comment:
Let's make these labels a var so there's less chance of copy-pasta mistakes.

```suggestion
	indicesLabels = []string{"index", "cluster"}
	indicesDocsPrimary = prometheus.NewDesc(
		prometheus.BuildFQName(namespace, "indices", "docs_primary"),
		"Count of documents with only primary shards",
		indicesLabels, nil,
	)
```